interface/src/app/signature/signature.js

```diff
@@ -23,8 +23,39 @@ angular.module('adage.signature', [
   });
 }])
 
-.controller('SignatureCtrl', ['Signature', '$stateParams', '$log',
-  function SignatureController(Signature, $stateParams, $log) {
+.component('participationTypeSelector', {
+  templateUrl: 'signature/participation-type-selector.tpl.html',
+  bindings: {
+    selectedParticipationType: '=',
+    onChange: '&'
+  },
+  controller: ['ParticipationType', '$log', function(ParticipationType, $log) {
+    var self = this;
+
+    // apiReturnLimit is what the 'limit' parameter will be set to in the
+    // ParticipationType resource's GET request. By setting it to 0, we
+    // are actually telling the Tastypie API that we want it to return ALL
+    // objects from this endpoint (i.e. that we do not want a maximum
+    // limit to how many objects it returns).
+    var apiReturnLimit = 0;
+
+    ParticipationType.get({limit: apiReturnLimit},
+      function success(response) {
+        self.availableParticipationTypes = response.objects;
+        self.selectedParticipationType = response.objects[0];
+      },
+      function error(response) {
+        var errMessage = errGen('Failed to get ParticipationTypes: ',
```

Comment:
No need to put `: ` at the end of the first argument of `errGen()`, whose implementation (in `app/utils.js`) will insert ": ".